static/app/components/discover/quickContextCommitRow.tsx

```diff
@@ -0,0 +1,94 @@
+import styled from '@emotion/styled';
+
+import UserAvatar from 'sentry/components/avatar/userAvatar';
+import CommitLink from 'sentry/components/commitLink';
+import {PanelItem} from 'sentry/components/panels';
+import TextOverflow from 'sentry/components/textOverflow';
+import {t, tct} from 'sentry/locale';
+import ConfigStore from 'sentry/stores/configStore';
+import space from 'sentry/styles/space';
+import {Commit} from 'sentry/types';
+
+import {formatCommitMessage} from '../commitRow';
+import ExternalLink from '../links/externalLink';
+
+interface QuickContextCommitRowProps {
+  commit: Commit;
+  customAvatar?: React.ReactNode;
+}
+
+function QuickContextCommitRow({commit, customAvatar}: QuickContextCommitRowProps) {
+  const user = ConfigStore.get('user');
+  const isUser = user?.id === commit.author?.id;
+  const hasPullRequestURL = commit.pullRequest && commit.pullRequest.externalUrl;
+
+  return (
+    <StyledPanelItem key={commit.id} data-test-id="quick-context-commit-row">
+      {customAvatar
+        ? customAvatar
+        : commit.author && (
+            <div>
+              <UserAvatar size={24} user={commit.author} />
+            </div>
+          )}
+
+      <CommitLinks>
+        {hasPullRequestURL && commit.message && (
+          <LinkToPullRequest data-test-id="quick-context-commit-row-pr-link">
+            <ExternalLink href={commit.pullRequest?.externalUrl}>
+              {formatCommitMessage(commit.message)}
+            </ExternalLink>
+          </LinkToPullRequest>
+        )}
+        <LinkToCommit
+          prTitle={hasPullRequestURL && commit.message}
+          data-test-id="quick-context-commit-row-commit-link"
+        >
+          {tct('View commit [commitLink] by [author]', {
+            author: isUser ? t('You') : commit.author?.name ?? t('Unknown author'),
+            commitLink: (
+              <CommitLink
+                inline
+                showIcon={false}
+                commitId={commit.id}
+                repository={commit.repository}
+              />
+            ),
+          })}
+        </LinkToCommit>
+      </CommitLinks>
+    </StyledPanelItem>
+  );
+}
+
+const StyledPanelItem = styled(PanelItem)`
+  display: flex;
+  align-items: center;
+  gap: ${space(1)};
+  padding: 0;
+  border: none;
+
+  & + & {
+    margin-top: ${space(1)};
+  }
+`;
+
+const CommitLinks = styled('div')`
+  flex: 1;
+  flex-direction: column;
+  min-width: 0;
+`;
+
+const LinkToPullRequest = styled(TextOverflow)`
+  font-size: ${p => p.theme.fontSizeLarge};
+  line-height: 1.2;
+`;
+
+const LinkToCommit = styled(TextOverflow)<{prTitle: string | null | undefined}>`
+  font-size: ${p => (p.prTitle ? p.theme.fontSizeSmall : p.theme.fontSizeLarge)};
+  color: ${p => (p.prTitle ? p.theme.subText : p.theme.textColor)};
```

Comment:
I'd prefer to see `hasPrTitle` here because when you're looking at the component's call site you're not expecting this to somehow render the PR title. If it's a `has` kind of variable name then it better communicates that it's going to show one of two states

Reply:
Yeah agree, pushed the 'has' prefix.